blackbox/docs/general/builtins/aggregation.rst

```diff
@@ -13,33 +13,91 @@ Aggregation
 Introduction
 ============
 
-An *aggregation function* operates on multiple values of the specified column
-of your ``SELECT``. For example, ``SELECT avg(temperature)`` will return the
-average value of the ``temperature`` column across all rows being considered.
+An **aggregate function** computes a single result from a set of input values.
 
-You can apply aggregation functions to the results of a whole query, or to each
-of the rows produced by a :ref:`sql_dql_group_by` query.
+::
+
+   cr> SELECT count(*)
+   ... FROM locations;
+   +----------+
+   | count(*) |
+   +----------+
+   |       13 |
+   +----------+
+   SELECT 1 row in set (... sec)
+
+In the example above the ``count(*)`` aggregate function computes the result
+across all rows.
+
+Aggregate functions can be used with the :ref:`sql_dql_group_by` clause of
+the :ref:`sql_reference_select` statement. If so, an aggregate function computes
+a single result per each group of input values produced by a query.
+
+::
 
-So for example, ``GROUP BY type, location`` will produce one result row for
-every distinct combination of type and location. In this instance,
-``avg(temperature)`` would return the average temperature for every grouped
-row.
+   cr> SELECT kind, count(*)
+   ... FROM locations
+   ... GROUP BY kind;
+   +-------------+----------+
+   | kind        | count(*) |
+   +-------------+----------+
+   | Galaxy      |        4 |
+   | Star System |        4 |
+   | Planet      |        5 |
+   +-------------+----------+
+   SELECT 3 rows in set (... sec)
 
-If you're not using `GROUP BY`, an aggregation function will collapse your
-query result into one row, with one returned aggregation value.
+For a tabulated summary of aggregate functions, see :ref:`sql_dql_aggregation`.
 
-For a tabulated summary of aggregation functions, see
-:ref:`sql_dql_aggregation`.
+.. _aggregate-expressions:
+
+Aggregate Expressions
+=====================
+
+An aggregate expression represents the application of an aggregate function
+across rows selected by a query. Besides the function signature, expressions
+might contain supplementary clauses and keywords.
+
+Synopsis
+--------
+
+The synopsis of an aggregate expression is one of the following
+
+::
+
+   aggregate_function ( * ) [ FILTER ( WHERE condition ) ]
+   aggregate_function ( [ DISTINCT ] expression [ , ... ] ) [ FILTER ( WHERE condition ) ]
+
+where ``aggregate_function`` is a name of an
+:ref:`aggregate function <aggregate-functions>`.
+
+and ``expression`` is a column reference, scalar function or literal.
+
+If ``FILTER`` is specified, then only the rows that met the
+:ref:`sql_dql_where_clause` condition are supplied to the aggregate function.
+
+The optional ``DISTINCT`` keyword is not fully supported in aggregate
```

Comment:
This together with the limitations reads a bit strange as afaik distinct is only supported on a few aggregates(count, avg) where it is explicitly documented. Imho the limitation point is also written falsely as it reads like the limitation only applies to versions <= 2.0.0.  Maybe we can write it here like:

"The optional ``DISTINCT`` keyword is only supported by aggregation function which explicitly mentioning the support for it"

And remove the limitation point as documenting older version behaviour doesn't make sense to me cause our whole documentation is already versioned.